browser/importer/brave_profile_writer.cc

```diff
@@ -441,3 +447,72 @@ void BraveProfileWriter::UpdateWindows(
   if (active)
     active->window()->Show();
 }
+
+
+// NOTE: the strings used as keys match the values found in Muon:
+// browser-laptop/js/data/searchProviders.js
+std::map<std::string,TemplateURLPrepopulateData::PrepopulatedEngine>
+    importable_engines = {
+      {"Amazon", TemplateURLPrepopulateData::amazon},
+      {"Bing", TemplateURLPrepopulateData::bing},
+      {"DuckDuckGo", TemplateURLPrepopulateData::duckduckgo},
+      {"Ecosia", TemplateURLPrepopulateData::ecosia},
+      {"GitHub", TemplateURLPrepopulateData::github},
+      {"Google", TemplateURLPrepopulateData::google},
+      {"Infogalactic", TemplateURLPrepopulateData::infogalactic},
+      {"MDN Web Docs", TemplateURLPrepopulateData::mdnwebdocs},
+      {"Qwant", TemplateURLPrepopulateData::qwant},
+      {"searx", TemplateURLPrepopulateData::searx},
+      {"Semantic Scholar", TemplateURLPrepopulateData::semanticscholar},
+      {"Stack Overflow", TemplateURLPrepopulateData::stackoverflow},
+      {"StartPage", TemplateURLPrepopulateData::startpage},
+      {"Twitter", TemplateURLPrepopulateData::twitter},
+      {"Wikipedia", TemplateURLPrepopulateData::wikipedia},
+      {"Wolfram Alpha", TemplateURLPrepopulateData::wolframalpha},
+      {"Yahoo", TemplateURLPrepopulateData::yahoo},
+      {"Yandex", TemplateURLPrepopulateData::yandex},
+      {"YouTube", TemplateURLPrepopulateData::youtube}
+    };
+
+void BraveProfileWriter::UpdateSettings(const SessionStoreSettings& settings) {
+  int default_search_engine_id =
+      TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_INVALID;
+
+  // Set the default search engine
+  TemplateURLService* url_service =
+      TemplateURLServiceFactory::GetForProfile(profile_);
+  if (url_service) {
+    auto it = importable_engines.find(settings.default_search_engine);
+    if (it != importable_engines.end()) {
+      TemplateURLPrepopulateData::PrepopulatedEngine engine = it->second;
+      std::unique_ptr<TemplateURLData> template_data =
+          TemplateURLDataFromPrepopulatedEngine(engine);
+      default_search_engine_id = engine.id;
+      LOG(INFO) << "Setting default search engine to "
+          << settings.default_search_engine;
+      TemplateURL provider_url(*template_data);
+      url_service->SetUserSelectedDefaultSearchProvider(&provider_url);
+    }
+  }
+
+  // Save alternate engine (for private tabs) to preferences
+  PrefService* prefs = profile_->GetPrefs();
+  if (prefs) {
+    prefs->SetBoolean(kUseAlternativeSearchEngineProvider,
+        settings.use_alternate_private_search_engine);
+
+    // Provider for Tor tabs
+    if (settings.use_alternate_private_search_engine_tor) {
+      // if enabled, set to either Qwant or DuckDuckGo (based on region)
+      int private_search_engine_id = brave::IsRegionForQwant(profile_)
+        ? TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_QWANT
+        : TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_DUCKDUCKGO;
```

Comment:
The above logic is a little fragile (could break if we switch our defaults); There isn't an existing way to get the "private search" ID that I could find

Reply:
Fixed with https://github.com/brave/brave-core/pull/1064/commits/d3d4a41f9ba001a36524feb4496464b794c4af8e